pkg/network/ebpf/c/prebuilt/tracer.c

```diff
@@ -103,6 +111,42 @@ int kretprobe__tcp_sendmsg(struct pt_regs *ctx) {
     return handle_message(&t, sent, 0, CONN_DIRECTION_UNKNOWN, packets_out, packets_in, PACKET_COUNT_ABSOLUTE, skp);
 }
 
+SEC("kprobe/tcp_recvmsg")
+int kprobe__tcp_recvmsg(struct pt_regs *ctx) {
```

Comment:
You should use a `kretprobe` here since the return value from `tcp_recvmsg` could be `<= 0`.

Reply:
Calling `handle_message` with `0` counts is going to also add empty entries if the `tcp_recvmsg` returns `<= 0`